src/classes/TG_ValidateData_TDTM.cls

```diff
@@ -0,0 +1,411 @@
+        
+                }
+               
+            }
+        
+        
+            if (findCourseConnectionList != NULL && findCourseConnectionList.size() > 0) {
+            
+                findCourseConnection(findCourseConnectionList, termGradeContacts, termGradeCourseOffering, contactTermGradeMap);
+        
+            }
+        
+        
+            if (checkCourseContactList != NULL && checkCourseContactList.size() > 0) {
+            
+                validateCourseContact(checkCourseContactList);
+        
+            }       
+        
+        
+            if (findContactList != NULL && findContactList.size() > 0) {
+                                
+                findContact(findContactList);
+            }
+        
+        
+            if (findCourseOfferingList != NULL && findCourseOfferingList.size() > 0) {
+        
+                findCourseOffering(findCourseOfferingList);     
+        
+            }
+        
+        
+            if (findContactCourseOfferingList != NULL && findContactCourseOfferingList.size() > 0) {
+            
+                findContactCourseOffering(findContactCourseOfferingList);
+        
+            }
+        
+       }
+    
+        return NULL;
+
+    }
+    
+    /*******************************************************************************************************
+    * @description Queries for a Course Connection for the Term Grade based on the Contact and Course Offering.
+    * Updates the Term Grade if a Course Connection is found and throws an error if no Course Connection is found
+    * or more than 1 Course Connections are found.
+    * @param termGrades A List of Term Grades with no Course Connection.
+    ********************************************************************************************************/
+    
+    private void findCourseConnection(List<Term_Grade__c> termGrades, List<Id> termGradeContacts, List<Id> termGradeCourseOffering, Map<Id, List<Term_Grade__c>> contactTermGradeMap) {
+            
+        Map<Id, List<Term_Grade__c>> courseOfferTermGrade = new Map<Id, List<Term_Grade__c>>();
+        Map<Term_Grade__c, List<Course_Enrollment__c>> termGradeEnrollmentMap = new Map<Term_Grade__c, List<Course_Enrollment__c>>();
+        Map<Term_Grade__c, List<Course_Enrollment__c>> termGradeEnrollmentTermMap = new Map<Term_Grade__c, List<Course_Enrollment__c>>();
+    
+        for (Course_Enrollment__c eachEnrollment: [SELECT Id, Contact__c, Course_Offering__c FROM Course_Enrollment__c WHERE Contact__c IN:termGradeContacts AND Course_Offering__c IN:termGradeCourseOffering]) { 
+        
+            if (eachEnrollment != NULL){ 
+                
+                for(Term_Grade__c tg : contactTermGradeMap.get(eachEnrollment.Contact__c)) {
+            
+                    if (eachEnrollment.course_offering__c == tg.course_offering__c) {
+                
+                        if(termGradeEnrollmentMap.containsKey(tg)) {
+
```

Comment:
I really do like whitespace, but I'm not sure we need white space after the if()/else() stuff unless the first block of code within the if is another control structure like another if or a for loop, etc.